src-v2/Judge/NTS.Judge.Blazor/Pages/Dashboard/Components/Actions/EliminationForms/WithdrawForm.razor

```diff
@@ -19,20 +19,20 @@
     [Parameter]
     public Withdrawn? Withdrawn { get; set; } = default!;
     [Parameter]
-    public EventCallback OnSubmission { get; set; }
+    public EventCallback OnRestore { get; set; }
+    [Parameter]
+    public EventCallback OnEliminate { get; set; }
 
-    public async Task Revoke()
+    public async Task Eliminate()
     {
         Participation? participation = _participationBehind.SelectedParticipation;
         var tandemNumber = participation!.Tandem.Number;
-        await _participationBehind.RevokeQualification(tandemNumber, QualificationRevokeType.Withdraw);
+        await _participationBehind.Withdraw(tandemNumber);
+        await OnEliminate.InvokeAsync();
     }
 
     public async Task Restore()
```

Comment:
It's getting much better, but those hooks and methods have identical signature and usage across all elimination forms. What can we do about (except merging the forms, obviously)?